README.md

```diff
@@ -1,4 +1,274 @@
+
+This library provides **one** effortless logic for the reification of embedded DSLs in Scala. This comes along many **advantages**:
+
+- no knowledge of the [Reflection API](http://en.wikipedia.org/wiki/Reflection_%28computer_programming%29)
+- no overloading resolution
+- no dependence with types and count of arguments
+- no verbosity
+- no duplication of code
+
+Our solution makes use of [annotations](http://docs.scala-lang.org/overviews/reflection/annotations-names-scopes.html) and [macros](http://www.scala-lang.org/api/2.11.0/scala-reflect/index.html#scala.reflect.macros.blackbox.Context) to achieve this goal.
+
+##Table of contents:
+
+1. [Overview of the concept](#overview)
+2. [Details](#details)
+3. [Project Structure](#project-structure)
+4. [Usage](#usage)
+5. [References](#references)
+6. [Licence](#licence)
+
+--
+
+# Overview
+Writing embedding DSLs implies to go though the difficult task of reification. 
+Reification means when a function *take(x: Int): Query[T]* is written in Scala it will be converted to a corresponding DSL object *Take*. *Take* is the Intermediate Representation (IR) of the function *take()*. The naive method would code conditional statements in order to **identify** which function has been called. Identifying is strenuous and redundant because it must check many criteria, take in account aspects such as overriding of the function and this for all functions or classes. It is not a best practices and it creates as much logics of reification than there is cases to reify.
+Hopefully, this can be facilitated by libraries and DirectEmbedding is one of them.
+
+*[More details in  Quasiquotes](#quasiquotes)*
+
+### Attach Metadata
+The idea developed by DirectEmbedding is to attach metadata about the corresponding IR and thus it gets rid of the demanding task of identification. It becomes straight forward to solve the previous issues. When it was required to verify the name of the function, its types, its types arguments, its arguments count, now this is declared along the function. Overriding? It is also very simple, it simply needs to attach the corresponding IR.
+
+### Annotate
+Scala annotations can attribute an object to declarations. Thus, annotations permit to attach metadata, that is to say, the IR. The annotation is accessible through the symbol of the AST.
+
+### Reify
+DirectEmbedding reifies the DSL while the compilation. This means that the given AST for an annotated function will be reified into the DSL representation at compile-time. In order to do so, there is the need to *pattern match* all the different ASTs that can be generated by an annotated declaration. Once the arguments, the types arguments and the symbol of the annotated code are extracted from the AST then the reification occurs via a macro. The macro applies the arguments and types arguments to the obtained IR from the symbol. The macro returns the reified tree i.e. the IR with the arguments and types arguments correctly applied. 
```

Comment:
"the types arguments" -> the type arguments

I do not understand the pattern match part.